NIFI-9378 Create new artifact in nifi-assembly that packages all exte…

bbende opened the pull request:
…nsion manifests.

This produces a new artifact during building of `nifi-assembly` - `nifi-1.16.0-SNAPSHOT-manifests.zip`. This must be done from `nifi-assembly` since that is where the determination is made about which NARs to include in the assembly. 

This zip file contains all of the `META-INF/docs/extension-manifest.xml` files from all NARs in the assembly. It is also attached to the build so it will be published as a Maven artifact on a release. This allows creating a dependency on this artifact with the following:
```
<dependency>
    <groupId>org.apache.nifi</groupId>
    <artifactId>nifi-assembly</artifactId>
    <version>1.16.0-SNAPSHOT</version>
    <classifier>manifests</classifier>
    <type>zip</type>
</dependency>
```

rfellows commented:
Reviewing...

rfellows commented on nifi-assembly/pom.xml:
```diff
@@ -51,6 +71,30 @@ language governing permissions and limitations under the License. -->
+                                <descriptor>src/main/assembly/extension-manifests.xml</descriptor>
+                            </descriptors>
+                            <tarLongFileMode>posix</tarLongFileMode>
+                            <formats>
+                                <format>dir</format>
+                                <format>zip</format>
+                            </formats>
```
Should the formats be defined in the assembly descriptor file instead?

bbende replied:
I don't have a strong opinion here, but the reason I put them here and not in the descriptor file was because the other existing execution above this (`make shared resource`) already defines them in the pom, so it seemed consistent. That one was done through the pom config because there are profiles that alter the formats like "dir-only" to save time, but for this case we wouldn't need that so it could be fine to put it in the descriptor.

rfellows commented on nifi-assembly/src/main/assembly/extension-manifests.xml:
```diff
@@ -0,0 +1,28 @@
+    <id>manifests</id>
+    <includeBaseDirectory>true</includeBaseDirectory>
+    <baseDirectory>nifi-${project.version}-manifests</baseDirectory>
+
```
Should define the format here. Also, I think including the `dir` format is redundant. e can probably do without it.
```suggestion

    <formats>
        <format>zip</format>
    </formats>

```

rfellows reviewed (commented):
Just a minor suggestion...

bbende commented:
@rfellows thanks for reviewing, just pushed up a commit that makes a few tweaks..
- Flattened the directory structure to remove the META-INF/docs
- Removed the zip format
- Removed the version from base dir so a consumer can just expect a constant dir of nifi-manifests

Left the format in the pom for consistency with the other assembly execution.

rfellows reviewed (approved):
The latest changes look good! Thanks @bbende!
